### ISSUE TITLE
Analyse how to improve Luigi Core unit tests

### ISSUE THREAD

**Issue by jesusreal** [SAP/luigi#551]
*Description*
This tickets is the result of some discussions generated by https://github.com/SAP/luigi/issues/289.

Before we improve the unit tests for Luigi Core we need to analyse how to improve the tests, so this ticket will serve as analysis and then from our conclusions we will create follow up tasks to do the actual improvements.

*Expected result*
The result of this ticket should be a list of action items that can be evalutated and eventually translated to tasks.

*Timeboxed to maximum 1 working day*

**Comment by jesusreal** [comment]
As conclusion from my research, below I am listing the actions items I would execute. They need to be discussed and accepted before working on them:

1. Make the whole codebase testable. Export one object per file, instead of exporting each function from the file individually (affects mostly to helpers and services). Otherwise, those exported functions cannot be mocked while writing unit tests for functions calling them.

2. There are existing unit tests that are not actually unit tests because calls to other functions are not mocked. An example of that are the unit tests for _handleRouteChange_ method. Once step 1 is done, we can proceed with this.

3. Not possible to unit test Svelte components logic with current setup:
  - Short term solution: export logic to external files and write unit tests for that logic. An additional advantage of that is that components will be cleaner.
  - Long term solution *(to be decided, as it implies using a different unit testing framework)*: use Cypress for unit testing (we use it currently already for e2e tests). Svelte components can be tested thanks to [cypress-svelte-unit-test](https://www.npmjs.com/package/cypress-svelte-unit-test) package.
  - Main contributor states [here](https://www.infoq.com/news/2019/05/svelte-3-interview-rich-harris/) that unit testing story is work in progress:
![Screenshot 2019-06-03 at 15.00.18.png](https://images.zenhubusercontent.com/5b90f41c0e23a90b8f9c4f5d/d46e8865-9c9a-4684-889c-8b0736cd20d9)
4. Write tests for Luigi client as well - nice to have, but not the 1st priority
